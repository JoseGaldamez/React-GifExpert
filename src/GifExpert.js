import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpert = ( {defaultCategory = [] } ) => {
    
    const [categories, setcategories] = useState(defaultCategory);
    
    return (
        <>
            <h2>Gif Expert</h2>
            <AddCategory setCategories={setcategories} />
            <hr />
            <div>
                <ol>
                    {
                        categories.map((category, index) => (
                            <GifGrid key={category} category={category} />
                         ) )
                    }
                </ol>
            </div>
        </>
    )
}


export default GifExpert;
