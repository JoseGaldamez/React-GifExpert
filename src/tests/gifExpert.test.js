import React from "react";

import { shallow } from "enzyme";
import GifExpert from "../GifExpert";

describe("Revisando GifexpertApp", () => {

    test("Debe mostrarse correctamente", ()=>{
        const wrapper = shallow(<GifExpert />);
        expect(wrapper).toMatchSnapshot();
    });

    test("Debe mostrar una lista de categorias", ()=>{
        const categories = ['One punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpert defaultCategory={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
});