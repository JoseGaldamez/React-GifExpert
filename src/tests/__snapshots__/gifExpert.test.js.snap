// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Revisando GifexpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <ol>
      <GifGrid
        category="One punch"
        key="One punch"
      />
      <GifGrid
        category="Dragon Ball"
        key="Dragon Ball"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`Revisando GifexpertApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <ol />
  </div>
</Fragment>
`;
